raiden/tests/unit/transfer/mediated_transfer/test_mediation_fee.py

```diff
@@ -560,14 +620,27 @@ def test_mfee4():
         amount_after_fees=PaymentAmount(amount), channels=[channels, channels]
     )
     assert fee_calculation
+
     amount_with_margin = calculate_safe_amount_with_fee(
-        fee_calculation.amount_without_fees, FeeAmount(sum(fee_calculation.mediation_fees))
+        amount, FeeAmount(sum(fee_calculation.mediation_fees))
     )
-    assert amount_with_margin == 555_452_155_494_633_177
+
+    # Calculate mediation fees for both mediators
+    med_fees = []
+    incoming_amount = amount_with_margin
+    for _ in range(2):
+        outgoing_amount = get_amount_without_fees(
+            amount_with_fees=incoming_amount, channel_in=channels[0], channel_out=channels[1]
+        )
+        assert outgoing_amount
+        med_fees.append(incoming_amount - outgoing_amount)
+        incoming_amount = outgoing_amount
+
+    assert amount_with_margin == 543_503_066_141_505_551
 
     # print values for scenario
     print("{:_} {:_}".format(deposit - amount_with_margin, deposit + amount_with_margin))
-    for med_fee in running_sum(fee_calculation.mediation_fees):
+    for med_fee in running_sum(med_fees):
```

Comment:
Why are these print statements here? from before? What is their purpose in a unit test?

Reply:
They provide a way to check the unittest against the corresponding scenario. Scenarios are hard to debug and this can speed up the debugging process a lot.

Reply:
Hmmm? What do you mean? This is a unit test not a scenario.I probably misunderstand something.

Reply:
This unittest should print exactly the numbers that are in the mfee4 scenario. The `assert amount_with_margin == 543_503_066_141_505_551` above mirrors the assert on the first channel in the scenario. When this unittest fails, we don't have to waste time on running and debugging the mfee4 scenario and when it passes but the scenario fails, we should look for issues caused by the integration (transport, service, configuration).

We could put the same amount of asserts as in the scenario here, but would not provide much benefit over asserting the `amount_with_margin`. Using the prints, we can also get the precise numbers for the scenario, which are mostly impossible to calculate manually for the mfee4 scenario. It is bad to use the code output as test case, but for this complicated case I could not find another feasible solution. This way the scenario at least works as a snapshot test.

Reply:
Sounds like you have a valid reason for them here. I will probably forget and bother you again in the future if I see it though :)